### ISSUE TITLE
Java Source Lang causes NPE

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-extras#538]
```
Caused by: java.lang.NullPointerException
        at org.eclipse.jdt.internal.compiler.CompilationResult.<init>(CompilationResult.java:101)
        at org.eclipse.jdt.internal.compiler.Compiler.internalBeginToCompile(Compiler.java:829)
        at org.eclipse.jdt.internal.compiler.Compiler.beginToCompile(Compiler.java:394)
        at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:444)
        at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:426)
        at org.eclipse.xtext.java.resource.JavaDerivedStateComputer.installFull(JavaDerivedStateComputer.java:279)
        at org.eclipse.xtext.java.resource.JavaResource.lambda$installFull$1(JavaResource.java:195)
        at org.eclipse.xtext.java.resource.JavaResource.initializing(JavaResource.java:205)
        at org.eclipse.xtext.java.resource.JavaResource.installFull(JavaResource.java:199)
        at org.eclipse.xtext.java.resource.JavaResource.getContents(JavaResource.java:168)
        at org.eclipse.xtext.build.IncrementalBuilder$InternalStatefulIncrementalBuilder.lambda$launch$2(IncrementalBuilder.java:264)
```

`org.eclipse.jdt.internal.compiler.Compiler.compile(ICompilationUnit[])`
is called with nulls in the list

**Comment by cdietrich** [comment]
for some reasons we have java resources with compilation unit null

**Comment by cdietrich** [comment]
![Bildschirmfoto vom 2019-12-03 08-38-19](https://user-images.githubusercontent.com/238736/70029864-4edcbe80-15a8-11ea-9228-b9ea919aba53.png)

**Comment by cdietrich** [comment]
workaround is not having `package-info.java`s

**Comment by szarnekow** [comment]
Probably also true for module-info

**Comment by cdietrich** [comment]
org.eclipse.xtext.java.resource.JavaDerivedStateComputer.isInfoFile(Resource)
might be missing at some places

**Comment by cdietrich** [comment]
here is a unit test

```
    @Test def void testPackageInfo() {
        val rs = resourceSet( 'a/package-info.java' -> '''
            package a;
        ''', 'a/MyClass.java' -> '''
        	package a;
        	public class MySuperClass {
        	}
        ''')
        for (r : rs.resources) {
        	println("r" +r )
        	for (o : r.contents) {
        	}
        }
    }
```

**Comment by cdietrich** [comment]
fixed in 2.21

**Comment by cdietrich** [comment]
can confirm that the sample project builds again with 2.21.0-SNAPSHOT
i also see the problem seems to happen depending on the order of compilation/loading
